cmd/testwrapper/testwrapper.go

```diff
@@ -28,16 +28,19 @@ import (
 
 const maxAttempts = 3
 
+// testAttempt keeps track of the test name, outcome, logs, and if the test is flakey.
+// After running the tests, each testAttempt is written to a json file.
 type testAttempt struct {
-	name          testName
-	outcome       string // "pass", "fail", "skip"
+	Name          TestName `json:"testName,omitempty"`
+	Outcome       string   `json:"outcome,omitempty"` // "pass", "fail", "skip"
 	logs          bytes.Buffer
-	isMarkedFlaky bool // set if the test is marked as flaky
+	IsMarkedFlaky bool `json:"isMarkedFlakey,omitempty"` // set if the test is marked as flaky
 }
 
-type testName struct {
-	pkg  string // "tailscale.com/types/key"
-	name string // "TestFoo"
+// TestName keeps track of the test name and its package.
+type TestName struct {
```

Comment:
I tried to see where testName needed to become exported and I couldn't see it.

Are you planning to also export TestAttempt for downstream things that will parse the results?

Reply:
ah good catch, not exporting it then